airflow/contrib/operators/livy_operator.py

```diff
@@ -0,0 +1,175 @@
+        }
+
+        self._spark_params['proxy_user'] = kwargs.get('proxy_user')
+        self._spark_params['class_name'] = kwargs.get('class_name')
+        self._spark_params['jars'] = kwargs.get('jars')
+        self._spark_params['py_files'] = kwargs.get('py_files')
+        self._spark_params['files'] = kwargs.get('files')
+        self._spark_params['driver_memory'] = kwargs.get('driver_memory')
+        self._spark_params['driver_cores'] = kwargs.get('driver_cores')
+        self._spark_params['executor_memory'] = kwargs.get('executor_memory')
+        self._spark_params['executor_cores'] = kwargs.get('executor_cores')
+        self._spark_params['num_executors'] = kwargs.get('num_executors')
+        self._spark_params['archives'] = kwargs.get('archives')
+        self._spark_params['queue'] = kwargs.get('queue')
+        self._spark_params['name'] = kwargs.get('name')
+
+        self._livy_conn_id = livy_conn_id
+        self._polling_interval = polling_interval
+        self._timeout = timeout
+
+        self._livy_hook = None
+        self._batch_id = None
+        self._start_ts = None
+
+    def _init_hook(self):
+        if self._livy_conn_id:
+            if self._livy_hook and isinstance(self._livy_hook, LivyHook):
+                self.log.info("livy_conn_id is ignored when Livy hook is already provided")
+            else:
+                self._livy_hook = LivyHook(livy_conn_id=self._livy_conn_id)
+
+        if not self._livy_hook:
+            raise AirflowException("Unable to create LivyHook")
+
+    def execute(self, context):
+        self._init_hook()
+
+        self._batch_id = self._livy_hook.post_batch(**self._spark_params)
+
+        self._start_ts = LivyOperator.get_ts()
+
+        if self._polling_interval > 0:
+            self.poll_for_termination(self._batch_id)
+
+        return self._batch_id
+
+    def poll_for_termination(self, batch_id):
+        """
+        Pool Livy for batch termination.
+
+        :param batch_id: id of the batch session to monitor.
+        :type batch_id: int
+        """
+        state = self._livy_hook.get_batch_state(batch_id)
+        while state not in TERMINAL_STATES:
+            self.log.debug('Batch with id %s is in state: %s', batch_id, state.value)
+            if self._timeout > 0 and LivyOperator.get_ts() - self._start_ts > self._timeout:
+                # timeout enabled and expired
+                self.log.warning("Batch %s execution exceeded timeout. Last known state: %s",
+                                 batch_id, state.value)
+                self.kill()
+                raise AirflowException("Batch timeout reached")
+            sleep(self._polling_interval)
+            state = self._livy_hook.get_batch_state(batch_id)
+        self.log.info("Batch with id %s terminated with state: %s", batch_id, state.value)
+        if state != BatchState.SUCCESS:
+            raise AirflowException("Batch did not succeed")
+
+    def on_kill(self):
```

Comment:
What is this method for? You only used `self.kill()` in the changes of this PR.

Reply:
@feluelle It's part of the BaseOperator API - this fn will be called by the scheduler when you "clear" a running task instance.

Reply:
Ah okay thank you @ashb :)